### PR TITLE
fix: hydration error on promise example

### PR DESCRIPTION
**Pull request by himself65**
This will fix the hydration issue. Not sure if this is correct

Fixes: https://github.com/dai-shi/waku/issues/202

### CODE REVIEW

File: examples/03_promise/src/components/Counter.tsx [hunk 1]
```diff
@@ -26,9 +26,12 @@ const Message = ({
...
-    <li>count: {count}</li>
-    <li>delayedMessage: {delayedMessage as any}</li>
-  </ul>
-);
+}) => {
+  const message = use(delayedMessage);
+  return (
+    <ul>
+      <li>count: {count}</li>
+      <li>delayedMessage: {message}</li>
```
Comment (dai-shi) [inline examples/03_promise/src/components/Counter.tsx:34]: Hm, ARAIU, React should internally `use` a promise which is directly put in JSX.

File: examples/03_promise/tsconfig.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
     "esModuleInterop": true,
     "module": "nodenext",
     "skipLibCheck": true,
+    "types": ["react/canary"],
```
Comment (dai-shi) [inline examples/03_promise/tsconfig.json:9]: Let's prefer `///` reference in the source code.

File: examples/03_promise/src/components/Counter.tsx (hunk 1 above)
Comment (himself65) [reply examples/03_promise/src/components/Counter.tsx:34]: I'm not sure why...🤔 maybe ask React member to get some help?

**Comment by dai-shi** [comment]
> This will fix the hydration issue. Not sure if this is correct

If using `use` fixes the issue, I think it's React's issue.

File: examples/03_promise/src/components/Counter.tsx (hunk 1 above)
Comment (dai-shi) [inline examples/03_promise/src/components/Counter.tsx:34]: I would like to merge this with something like this:
```suggestion
     {/* We want show the usage without `use` but it causes a hydration error. https://github.com/dai-shi/waku/issues/202 */}
      <li>delayedMessage: {use(delayedMessage)}</li>
```

**Review by dai-shi — APPROVED** [review]

**Pull request merged**